lib/bokken/accounts.ex

```diff
@@ -276,6 +276,31 @@ defmodule Bokken.Accounts do
     Repo.get!(Ninja, id) |> Repo.preload(preloads)
   end
 
+  @doc """
+  Gets a single ninja by its discord_id.
+
+  Raises `{:error, :not_found}` if the Ninja does not exist.
+
+  ## Examples
+
+      iex> get_ninja_by_discord("123")
+      {:ok, %Ninja{}}
+
+      iex> get_ninja_by_discord("456")
+      {:error, :not_found}
+  """
+  def get_ninja_by_discord(discord) do
+    social = %Bokken.Accounts.Social{name: :discord, username: discord}
+    ninjas = list_ninjas()
+    target = Enum.find(ninjas, fn ninja -> Enum.member?(ninja.socials, social) end)
```

Comment:
This will need improvements because it could be a gigantic list of ninjas and in that case (it is not, I know), we should not bring everything in memory. 

I think you could do something like:

```elixir
query = ~s|{"name": "discord", "username": "#{discord}"}|

from(p in Ninja, where: fragment("? <@ ANY(?)", ~s|#{query}|, p.socials))
|> Repo.one()
```

Reply:
That way, the database would do the work of finding the correct Ninja.

Learn more about JSONB querying at https://alchemist.camp/articles/querying-nested-map-ecto.

Reply:
This is really cool, I didn't know this existed. I'm gonna take a look into it 💪